crates/uv-auth/src/store.rs

```diff
@@ -0,0 +1,175 @@
+use lazy_static::lazy_static;
+use std::{collections::HashMap, sync::Mutex};
+
+use netrc::Authenticator;
+use tracing::warn;
+use url::Url;
+
+use crate::NetLoc;
+
+lazy_static! {
+    // Store credentials for NetLoc
+    static ref PASSWORDS: Mutex<HashMap<NetLoc, Option<Credential>>> = Mutex::new(HashMap::new());
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Credential {
+    Basic(BasicAuthData),
+    UrlEncoded(UrlAuthData),
+}
+
+impl Credential {
+    pub fn username(&self) -> String {
+        match self {
+            Credential::Basic(auth) => auth.username.clone(),
+            Credential::UrlEncoded(auth) => auth.username.clone(),
+        }
+    }
+    pub fn password(&self) -> Option<String> {
+        match self {
+            Credential::Basic(auth) => auth.password.clone(),
+            Credential::UrlEncoded(auth) => auth.password.clone(),
+        }
```

Comment:
I think it's a bit inflexible for these to clone. Callers can always clone if they need owned data, but callers that _don't_ need owned data are now _forced_ to clone. I think these should return references.

Reply:
Done